### PR TITLE
TS types

### PR DESCRIPTION
**Pull request by dforsber**
Initial take on TS types. Add `index.d.ts` types file and refer to in `package.json` as per general recommendation

### CODE REVIEW

**Comment by ali-habibzadeh** [comment]
We still need the possible column types for `enum PrestoClientPrestoTypes`

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    scheduled: boolean;
+    nodes: number;
+    totalSplits: number;
+    queuedSplits: number;
+    runningSplits: number;
+    completedSplits: number;
+  }
+
+  // NOTE: Needs to be extended with missing items
+  enum PrestoClientPrestoTypes {
```
Comment (ali-habibzadeh) [inline index.d.ts]: @tagomoris would you be able to provide the list for this?

**Comment by pwmcintyre** [comment]
any updates on this? it's a good feature!

**Comment by Amirault** [comment]
Where are we ?

**Comment by Amirault** [comment]
It will be cool to have this pr here : https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types

**Comment by tagomoris** [comment]
See #28 and please note your feedback and opinions there.

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    catalog: string;
+    schema: string;
+    timezone?: string;
+    info?: boolean;
+    cancel?: () => boolean;
+    state?: (error: any, query_id: string, stats: IPrestoClientStats) => void;
+    columns?: (error: any, data: IPrestoClientColumnMetaData[]) => void;
+    data?: (
+      error: any,
+      data: PrestoClientColumnDatum[],
```
Comment (Ekrekr) [inline index.d.ts:81]:
```suggestion
      data: IPrestoClientColumnDatum[],
```

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      cert: string;
+      ciphers?: string;
+      key?: string;
+      passphrase?: string;
+      secureProtocol: string;
+      pfx?: string;
+      rejectUnauthorized?: boolean;
+    };
+    port: number;
+    source: string;
```
Comment (Ekrekr) [inline index.d.ts:15]:
```suggestion
    port?: number;
    source?: string;
```

File: index.d.ts [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    user: string;
+    catalog: string;
+    checkInterval: number;
```
Comment (Ekrekr) [inline index.d.ts:18]:
```suggestion
    checkInterval?: number;
```

File: index.d.ts [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    name: string;
+    type: PrestoClientPrestoTypes;
+  }
+
+  type PrestoClientColumnDatum = [string, any];
+
+  interface IPrestoClientExecuteOptions {
+    query: string;
+    catalog: string;
+    schema: string;
```
Comment (Ekrekr) [inline index.d.ts:73]:
```suggestion
    catalog?: string;
    schema?: string;
```

**Review by Ekrekr — COMMENTED** [review]
Noticed some things should be optional where they weren't; nice to have them written down somewhere whichever direction the type declarations goes in.

File: index.d.ts [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    timezone?: string;
+    info?: boolean;
+    cancel?: () => boolean;
+    state?: (error: any, query_id: string, stats: IPrestoClientStats) => void;
+    columns?: (error: any, data: IPrestoClientColumnMetaData[]) => void;
```
Comment (Ekrekr) [inline index.d.ts:78]:
```suggestion
    columns?: (error: any, columns: IPrestoClientColumnMetaData[]) => void;
```
to make consistent with the name used in `data`

File: index.d.ts (hunk 1 above)
Comment (Ekrekr) [reply index.d.ts:53]: Full list can be found here: https://prestodb.io/docs/current/language/types.html

**Comment by joelgarciajr84** [comment]
https://github.com/joelgarciajr84/presto-ts

**Comment by tagomoris** [comment]
See #28

**Pull request closed without merging**